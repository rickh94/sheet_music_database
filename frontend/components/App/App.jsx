import React, { Component } from 'react'

export default class App extends Component {
  render() {
    return (
      <div>
        <h1>I am an app</h1>
      </div>
    )
  }
}

