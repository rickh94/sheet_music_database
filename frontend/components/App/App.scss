.margin-default {
  margin: 1vh 5vw;
}

.no-margin-padding {
  margin: 0;
  padding: 0;
}

.alert-text {
  text-transform: none;
}

