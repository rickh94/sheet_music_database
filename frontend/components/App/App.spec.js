import React from 'react'
import { shallow } from 'enzyme'
import App from './App'
import '../../setupTests'

describe('App', () => {
  const wrapper = shallow(<App />)

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders the routes', () => {
    expect(wrapper.exists('Connect(Routes)')).toBeTruthy()
  })

  it('renders two Providers (Redux and Alert)', () => {
    expect(wrapper.find('Provider').length).toEqual(2)
  })
})
