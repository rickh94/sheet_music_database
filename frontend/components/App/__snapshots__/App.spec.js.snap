// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Provider
        offset="30px"
        position="bottom left"
        template={[Function]}
        timeout={3000}
        transition="fade"
        type="info"
        zIndex={100}
      >
        <BrowserRouter>
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/login"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/logout"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/register"
            />
            <Route
              component={[Function]}
              default={true}
            />
          </Switch>
        </BrowserRouter>
      </Provider>,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <BrowserRouter>
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/login"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/logout"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/register"
            />
            <Route
              component={[Function]}
              default={true}
            />
          </Switch>
        </BrowserRouter>,
        "offset": "30px",
        "position": "bottom left",
        "template": [Function],
        "timeout": 3000,
        "transition": "fade",
        "type": "info",
        "zIndex": 100,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/login"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/logout"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/register"
            />
            <Route
              component={[Function]}
              default={true}
            />
          </Switch>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/login"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/logout"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/register"
              />,
              <Route
                component={[Function]}
                default={true}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/login",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/logout",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/register",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "default": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Provider
          offset="30px"
          position="bottom left"
          template={[Function]}
          timeout={3000}
          transition="fade"
          type="info"
          zIndex={100}
        >
          <BrowserRouter>
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/login"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/logout"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/register"
              />
              <Route
                component={[Function]}
                default={true}
              />
            </Switch>
          </BrowserRouter>
        </Provider>,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <BrowserRouter>
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/login"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/logout"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/register"
              />
              <Route
                component={[Function]}
                default={true}
              />
            </Switch>
          </BrowserRouter>,
          "offset": "30px",
          "position": "bottom left",
          "template": [Function],
          "timeout": 3000,
          "transition": "fade",
          "type": "info",
          "zIndex": 100,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/login"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/logout"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/register"
              />
              <Route
                component={[Function]}
                default={true}
              />
            </Switch>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/login"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/logout"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/register"
                />,
                <Route
                  component={[Function]}
                  default={true}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/login",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/logout",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/register",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "default": true,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
